includes/core-changes/corefx/3.0/fieldinfo-setvalue-exception.md

```diff
@@ -0,0 +1,38 @@
+### FieldInfo.SetValue throws exception for static, initialization-only fields
+
+Starting in .NET Core 3.0, an exception is thrown when you attempt to set a value on a static, initialization-only field by calling <xref:System.Reflection.FieldInfo.SetValue%2A?displayProperty=fullName>.
+
+#### Change description
+
+In .NET Framework and versions of .NET Core prior to 3.0, you could set the value of a static, initialization-only ([readonly in C#](~/docs/csharp/language-reference/keywords/readonly.md)) field by calling <xref:System.Reflection.FieldInfo.SetValue%2A?displayProperty=fullName>. However, setting a static, initialization-only field in this way resulted in unpredictable behavior based on the target framework and optimization settings.
+
+In .NET Core 3.0 and later versions, when you call <xref:System.Reflection.FieldInfo.SetValue%2A> on a static, initialization-only field, a <xref:System.FieldAccessException?displayProperty=nameWithType> exception is thrown.
+
+> [!TIP]
+> An initialization-only field is one that can only be set at the time it's declared or in the constructor for the containing class.
+
+#### Version introduced
+
+3.0
+
+#### Recommended action
+
+The best way to initialize static, initialization-only fields is in a static constructor. This applies to both dynamic and non-dynamic types. Or, remove the <xref:System.Reflection.FieldAttributes.InitOnly?displayProperty=nameWithType> attribute from the field, and then call <xref:System.Reflection.FieldInfo.SetValue%2A?displayProperty=nameWithType>.
```

Comment:
initialization-only -> init-only or InitOnly

init-only is the official name for this. It is name standardized by https://www.ecma-international.org/publications/standards/Ecma-335.htm . Also, it is the name we use in the API names: https://docs.microsoft.com/en-us/dotnet/api/system.reflection.fieldinfo.isinitonly?view=netcore-3.1 . I do not think we should be inventing a new name for it.